bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolEmuEMThingHandler.java

```diff
@@ -0,0 +1,299 @@
+            int diff = (int) (now - lastTime);
+            lastTime = now;
+
+            d.update(diff);
+        }
+    }
+
+    private void startAutomaticUpdate() {
+        ScheduledFuture<?> job = updateJob;
+        if (job == null || job.isCancelled()) {
+            updateJob = scheduler.scheduleWithFixedDelay(this::updateRunnable, 0, 1, TimeUnit.SECONDS);
+        }
+    }
+
+    private synchronized @Nullable ResolBridgeHandler getBridgeHandler() {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            logger.debug("Required bridge not defined for thing {}.", thing.getThingTypeUID());
+            return null;
+        } else {
+            return getBridgeHandler(bridge);
+        }
+    }
+
+    private synchronized @Nullable ResolBridgeHandler getBridgeHandler(Bridge bridge) {
+        ResolBridgeHandler bridgeHandler = null;
+
+        ThingHandler handler = bridge.getHandler();
+        if (handler instanceof ResolBridgeHandler) {
+            bridgeHandler = (ResolBridgeHandler) handler;
+        } else {
+            logger.debug("No available bridge handler found yet. Bridge: {} .", bridge.getUID());
+        }
+        return bridgeHandler;
+    }
+
+    private void registerResolThingListener(@Nullable ResolBridgeHandler bridgeHandler) {
+        if (bridgeHandler != null) {
+            bridgeHandler.registerResolThingListener(this);
+        } else {
+            logger.debug("Can't register {} at bridge as bridgeHandler is null.", this.getThing().getUID());
+        }
+    }
+
+    public int getVbusAddress() {
+        return vbusAddress;
+    }
+
+    public void useConnection(Connection connection) {
+        EmDeviceEmulator device = this.device;
+        if (device != null) {
+            device.stop();
+            device.removePropertyChangeListener(this);
+        }
+        device = new EmDeviceEmulator(connection, deviceId);
+        this.device = device;
+        device.addPropertyChangeListener(this);
+        device.start();
+        for (int i = 1; i <= 5; i++) {
+            setRelayChannelValue(i, device.getRelayValueByNr(i));
+        }
+        startAutomaticUpdate();
+    }
+
+    public void stop() {
+        EmDeviceEmulator device = this.device;
+        if (device != null) {
+            device.stop();
+        }
+        ScheduledFuture<?> updateJob = this.updateJob;
+        if (updateJob != null) {
+            updateJob.cancel(false);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String chID = channelUID.getId();
+        boolean update = false;
+        int channel = chID.charAt(chID.length() - 1) - '0';
+        float value = 0;
+        int intValue = 0;
+
+        if (command instanceof QuantityType<?>) {
+            value = Objects.requireNonNullElse(((QuantityType<?>) command).toUnit(SIUnits.CELSIUS),
+                    new QuantityType<>(888.8, SIUnits.CELSIUS)).floatValue();
```

Comment:
What is the meaning of the value of 888.8°C? You could log a warning if the unit cannot be converted. This occurs if the user sent an invalid command.

Reply:
888.8 is used in the VBUS in case of not available values - e. g. if the sensor wire would be broken. So in my eyes it fits quite well to the illegal input case here.

Reply:
Me as a user would be happy about a hint what went wrong. I leave it up to you how to handle that case.